ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -728,16 +1272,23 @@ def _checkConvObjective(self, traj, **kwargs):
       @ In, kwargs, dict, dictionary of parameters for convergence criteria
       @ Out, converged, bool, convergence state
     """
-    if len(self._optPointHistory[traj]) < 2:
-      return False
-    o1, _ = self._optPointHistory[traj][-1]
-    obj = o1[self._objectiveVar]
-    converged = (obj == self._convergenceCriteria['objective'])
-    self.raiseADebug(self.convFormat.format(name='objective',
-                                            conv=str(converged),
-                                            got=obj,
-                                            req=self._convergenceCriteria['objective']))
-
+    if len(self._objectiveVar) == 1: # This is for a single-objective Optimization case.
+      if len(self._optPointHistory[traj]) < 2:
+        return False
+      o1, _ = self._optPointHistory[traj][-1]
+      obj = o1[self._objectiveVar[0]]
+      converged = (obj == self._convergenceCriteria['objective'])
+      self.raiseADebug(self.convFormat.format(name='objective',
+                                              conv=str(converged),
+                                              got=obj,
+                                              req=self._convergenceCriteria['objective']))
+    else:                            # This is for a multi-objective Optimization case.
+      if len(self._optPointHistory[traj]) < 2:
+        return False
+      o1, _ = self._optPointHistory[traj][-1]
+      obj1 = o1[self._objectiveVar[0]]
+      obj2 = o1[self._objectiveVar[1]]
+      converged = (obj1 == self._convergenceCriteria['objective'] and obj2 == self._convergenceCriteria['objective'])
```

Comment:
These lines indicate the convergence only work for 2 objectives, I think you need to extend this part to handle multi-objectives optimizations with more than 2 objectives.